pandas/core/aggregation.py

```diff
@@ -197,3 +256,23 @@ def maybe_mangle_lambdas(agg_spec: Any) -> Any:
         mangled_aggspec = _managle_lambda_list(agg_spec)
 
     return mangled_aggspec
+
+
+def _relabel_result(result, reordered_result, func, columns, order):
+    """Internal function to reorder result if relabelling for dataframe.agg."""
+
+    reordered_indexes = [
+        pair[0] for pair in sorted(zip(columns, order), key=lambda t: t[1])
+    ]
+    idx = 0
+
+    # The reason is self._aggregate outputs different type of result if
+    # any column is only used once in aggregation
+    mask = isinstance(result, ABCDataFrame) and result.isna().any().any()
+
+    for col, fun in func.items():
```

Comment:
you are mutating the input argument, which IIUC is a ``DataFrame(index=columns)``; why don't you just create it in this function and return it?

This is a very hard to follow api the way you are doing it.

Reply:
sorry for late reply @jreback 

yeah, agree, this doesn't look optimal, and input is indeed `DataFrame(index=columns)`, the reason not to create it in this function is `DataFrame` is defined in `frame.py`, if we want to use it, then need to import from `frame.py`, and this function is also gonna be used in `frame.py`, which will result in the conflict imports.

Any suggestions on how to work around it?

Reply:
i remove the input arugment of `reordered_result` and return a dictionary in this method, and form into Dataframe in `frame.py` instead, is it a bit clearer?